docs/content/docs/ops/state/checkpoints.md

```diff
@@ -175,6 +175,35 @@ above).
 $ bin/flink run -s :checkpointMetaDataPath [:runArgs]
 ```
 
+### Checkpoint/Savepoint guarantees
+Not always the snapshot can be successfully restored. 
+The following list contains operations which can be done without recovery lose depends on the type of the snapshot.
+
+| Operation                                             | Canonical Savepoint | Native Savepoint         | Aligned Checkpoint       | Unaligned Checkpoint     |
+|:------------------------------------------------------|:--------------------|:-------------------------|:-------------------------|:-------------------------|
+| Statebackend change                                   | Supported           | NOT Supported            | NOT Supported            | NOT Supported            |
+| Self-contained and relocatable                        | Supported           | Supported                | NOT Supported            | NOT Supported            |
+| State Processor API(reading)                          | Supported           | Experimentally Supported | Experimentally Supported | NOT Supported            |
+| State Processor API(writing)                          | Supported           | NOT Supported            | NOT Supported            | NOT Supported            |
+| Schema evolution                                      | Supported           | Experimentally Supported | Experimentally Supported | Experimentally Supported |
+| Flink minor (1.x → 1.y) version upgrade               | Supported           | Supported                | Supported                | NOT Supported            |
+| Flink bug/patch (1.14.x → 1.14.y) version upgrade     | Supported           | Supported                | Supported                | Supported                |
+| Arbitrary job upgrade                                 | Supported           | Supported                | Supported                | NOT Supported            |
+| Job upgrade w/o changing graph shape and record types | Supported           | Supported                | Supported                | NOT Supported            |
+| Rescaling                                             | Supported           | Supported                | Supported                | Supported                |
```

Comment:
I think we should definitely avoid color for this table (more user-friendly and more accessible to people who might be color-blind, etc) but we could consider using an abbreviation or a symbol to represent "Supported", "NOT Supported", and "Experimentally Supported" to make the entire table less wordy and cramped.  

For example, we could use "✓", "x", "!" and then have a legend for that beforehand.